lib/util/Components.js

```diff
@@ -456,15 +456,15 @@ function componentRule(rule, context) {
       let scope = context.getScope();
       while (scope) {
         const node = scope.block;
-        const isClass = node.type === 'ClassExpression';
         const isFunction = /Function/.test(node.type); // Functions
-        const isArrowFunction = node.type === 'ArrowFunctionExpression';
-        let functionScope = scope;
+        const isArrowFunction = astUtil.isArrowFunction(node);
+        let enclosingScope = scope;
         if (isArrowFunction) {
-          functionScope = utils.getParentFunctionScope(scope);
+          enclosingScope = utils.getArrowFunctionScope(scope);
```

Comment:
this seems like it could be `const enclosingScope = isArrowFunction ? utils.getArrowFunctionScope(scope) : scope;`

Reply:
sure thing, fixed